packages/eslint-plugin/docs/rules/no-unnecessary-type-conversion.mdx

```diff
@@ -0,0 +1,74 @@
+---
+description: 'Disallow conversion idioms when they do not change the type or value of the expression.'
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-unnecessary-type-conversion** for documentation.
+
+JavaScript has several idioms used to convert values to certain types. With TypeScript, it is possible to see at build time if the value is already of that type, making the conversion unnecessary.
```

Comment:
[Docs] I think we'll want to explicitly call out what the things are in plain text, before getting to the explanation of why we don't want them -> examples of them.

```suggestion
JavaScript provides several commonly used idioms to convert values to a specific types:

* Primitive coercion (e.g. `Boolean(value)`, `String(value)`): using a built-in primitive function
* String concatenation (e.g. `value + ""`): turning a value into a string
* Unary coercion (e.g. `+value`, `!!value`): using a built-in operator

These conversions are unnecessary if the value is already that type.
```